README.md

```diff
@@ -2,24 +2,208 @@
+
+##### (Advanced) Example 2
+
+Revisiting the Shapes and Canvas example, the three toolbar buttons would result in 3 `onFunction` `Action` dispatchers, 3 `Actions`, and 3 `handleFunction` `Reducers`.
+
+- `onFunctions`
+
+        onCreateCircleToolbarButtonClicked
+        onCreateSquareToolbarButtonClicked
+        onCreateTriangleToolbarButtonClicked
+
+- Action Types:
+
+        CREATE_CIRCLE_TOOLBAR_BUTTON_CLICKED
+        CREATE_SQUARE_TOOLBAR_BUTTON_CLICKED
+        CREATE_TRIANGLE_TOOLBAR_BUTTON_CLICKED
+
+- `handleFunctions`
+
+        handleCreateCircleToolbarButtonClicked
+        handleCreateSquareToolbarButtonClicked
+        handleCreateTriangleToolbarButtonClicked
+
+Hopefully it will be obvious that one of these sets might be implemented thusly:
+
+The UI element defiend by this JSX
+
+    <button onClick={onCreateCircleToolbarButtonClicked}>Create Circle</button>
+
+Calls this `Action` Dispatcher
+
+    onCreateCircleToolbarButtonClicked()
+
+Which dispatches an `Action` of Type
+
+    CREATE_CIRCLE_TOOLBAR_BUTTON_CLICKED
+
+Which triggers this `Reducer` `handleFunction`:
+
+    handleCreateCircleToolbarButtonClicked()
+
+### Why We Did This
+
+There are several reasons for this convention. First, grouping the `Action` dispatcher, `Action` type constant,and the `Reducer` function close together ensures that should one of them need to have their names changed, the corresponding pieces are easy to find and resynchrnize. Further, the consistent naming convention will ease development and integration of community-supported components. Additionally, since all of the files related to a UI compnent live in a compartmentalized domain-driven folder structure, components are likely to be self-contained and easily dragged from one project to another.
+
+Finally, while React has implemented things such as the Context API, and Redux is often more power than might be necessary, this approach reduces learning barriers for new developers, by teaching a single consistent way for handling both local and global scopes, without having to learn two techniques, and the nuanced situations when to use each. Meanwhile, experienced developers and highly complex Runes are encouraged to employ whatever patterns and conventions are comfortable and efficient.
+
```

Comment:
We can shrink this. No need to defend ourselves just yet. We can leave probably about two sentences worth of “why we do this” and allow people to ask questions if there’s their own reason.